diff --git a/bin/antigen.zsh b/bin/antigen.zsh
index e3bf435..fd3de63 100644
--- a/bin/antigen.zsh
+++ b/bin/antigen.zsh
@@ -950,7 +950,6 @@ Commands:
   restore      Restore plugin state from a snapshot file.
   revert       Revert plugins to their state prior to the last time 'antigen
                update' was run.
-  selfupdate   Update antigen.
   snapshot     Create a snapshot of all active plugin repos and save it to a
                snapshot file.
   update       Update plugins.
@@ -1156,27 +1155,6 @@ antigen-revert () {
     return 1
   fi
 }
-# Update (with `git pull`) antigen itself.
-# TODO: Once update is finished, show a summary of the new commits, as a kind of
-# "what's new" message.
-antigen-selfupdate () {
-  (\cd -q $_ANTIGEN_INSTALL_DIR
-   if [[ ! ( -d .git || -f .git ) ]]; then
-     echo "Your copy of antigen doesn't appear to be a git clone. " \
-       "The 'selfupdate' command cannot work in this case."
-     return 1
-   fi
-   local head="$(git rev-parse --abbrev-ref HEAD)"
-   if [[ $head == "HEAD" ]]; then
-     # If current head is detached HEAD, checkout to master branch.
-     git checkout master
-   fi
-   git pull
-
-   # TODO Should be transparently hooked by zcache
-   antigen-reset &>> /dev/null
-  )
-}
 antigen-snapshot () {
   local snapshot_file="${1:-antigen-shapshot}"
   local urls url dir version_hash snapshot_content
@@ -1977,7 +1955,6 @@ _antigen () {
     'reset:Clears cache'
     'restore:Restore the bundles state as specified in the snapshot'
     'revert:Revert the state of all bundles to how they were before the last antigen update'
-    'selfupdate:Update antigen itself'
     'snapshot:Create a snapshot of all the active clones'
     'theme:Switch the prompt theme'
     'update:Update all bundles'
diff --git a/src/_antigen b/src/_antigen
index ab1d53c..9067435 100644
--- a/src/_antigen
+++ b/src/_antigen
@@ -14,7 +14,6 @@ _antigen () {
     'reset:Clears cache'
     'restore:Restore the bundles state as specified in the snapshot'
     'revert:Revert the state of all bundles to how they were before the last antigen update'
-    'selfupdate:Update antigen itself'
     'snapshot:Create a snapshot of all the active clones'
     'theme:Switch the prompt theme'
     'update:Update all bundles'
diff --git a/src/commands/help.zsh b/src/commands/help.zsh
index ab56806..c434336 100644
--- a/src/commands/help.zsh
+++ b/src/commands/help.zsh
@@ -20,7 +20,6 @@ Commands:
   restore      Restore plugin state from a snapshot file.
   revert       Revert plugins to their state prior to the last time 'antigen
                update' was run.
-  selfupdate   Update antigen.
   snapshot     Create a snapshot of all active plugin repos and save it to a
                snapshot file.
   update       Update plugins.
diff --git a/src/commands/self-update.zsh b/src/commands/self-update.zsh
deleted file mode 100644
index 6aa76a5..0000000
--- a/src/commands/self-update.zsh
+++ /dev/null
@@ -1,21 +0,0 @@
-# Update (with `git pull`) antigen itself.
-# TODO: Once update is finished, show a summary of the new commits, as a kind of
-# "what's new" message.
-antigen-selfupdate () {
-  (\cd -q $_ANTIGEN_INSTALL_DIR
-   if [[ ! ( -d .git || -f .git ) ]]; then
-     echo "Your copy of antigen doesn't appear to be a git clone. " \
-       "The 'selfupdate' command cannot work in this case."
-     return 1
-   fi
-   local head="$(git rev-parse --abbrev-ref HEAD)"
-   if [[ $head == "HEAD" ]]; then
-     # If current head is detached HEAD, checkout to master branch.
-     git checkout master
-   fi
-   git pull
-
-   # TODO Should be transparently hooked by zcache
-   antigen-reset &>> /dev/null
-  )
-}
diff --git a/tests/selfupdate.t b/tests/selfupdate.t
deleted file mode 100644
index f0576a3..0000000
--- a/tests/selfupdate.t
+++ /dev/null
@@ -1,73 +0,0 @@
-Set environment variables for this test case
-
-  $ TEST_DIR=$PWD
-  $ TEST_HOST=$TEST_DIR/host
-  $ TEST_NORMAL=$TEST_DIR/client
-  $ TEST_SUBMODULE=$TEST_DIR/submodule
-
-Create fake host repository
-
-  $ mkdir -p $TEST_HOST
-  $ cd $TEST_HOST
-  $ git init
-  Initialized empty Git repository in * (glob)
-  $ git config user.name 'test'
-  $ git config user.email 'test@test.test'
-  $ echo 1 > ver
-  $ git add ver
-  $ git commit -m "1"
-  [master (root-commit) ???????] 1 (glob)
-   1 file changed, 1 insertion(+)
-   create mode 100644 ver
-
-Create a normal repository cloning from host
-
-  $ git clone $TEST_HOST $TEST_NORMAL &> /dev/null
-
-Create a submodule repository cloning from host
-
-  $ mkdir -p $TEST_SUBMODULE
-  $ cd $TEST_SUBMODULE
-  $ git init
-  Initialized empty Git repository in * (glob)
-  $ git config user.name 'test'
-  $ git config user.email 'test@test.test'
-  $ git submodule add $TEST_HOST antigen &> /dev/null
-  $ git commit -m "1"
-  [master (root-commit) ???????] 1 (glob)
-   2 files changed, 4 insertions(+)
-   create mode 100644 .gitmodules
-   create mode 160000 antigen
-
-Update host repository
-
-  $ cd $TEST_HOST
-  $ echo 2 > ver
-  $ git add ver
-  $ git commit -m "2"
-  [master ???????] 2 (glob)
-   1 file changed, 1 insertion(+), 1 deletion(-)
-
-Use selfupdate from normal repository
-
-  $ _ANTIGEN_INSTALL_DIR=$TEST_NORMAL antigen-selfupdate
-  From * (glob)
-     ???????..???????  master     -> origin/master (glob)
-  Updating ???????..??????? (glob)
-  Fast-forward
-   ver |*2 +- (glob)
-   1 file changed, 1 insertion(+), 1 deletion(-)
-  $ _ANTIGEN_INSTALL_DIR=$TEST_NORMAL antigen-selfupdate
-  Already up-to-date.
-
-Use selfupdate from submodule repository
-
-  $ _ANTIGEN_INSTALL_DIR=$TEST_SUBMODULE/antigen antigen-selfupdate
-  From * (glob)
-     ???????..???????  master     -> origin/master (glob)
-  Updating ???????..??????? (glob)
-  Fast-forward
-   ver |*2 +- (glob)
-   1 file changed, 1 insertion(+), 1 deletion(-)
-  $ _ANTIGEN_INSTALL_DIR=$TEST_SUBMODULE/antigen antigen-selfupdate
-  Already up-to-date.
